Make clock-setup safe for s390

On s390 D-I is not allowed to change the system clock. However, clock-setup
is also responsible for running tzsetup, which we do want on s390.

Make clock-setup safe for s390 by only running tzsetup in the postinst and
excluding the finish-install script from the udeb.

diff --git a/packages/clock-setup/debian/clock-setup.postinst b/packages/clock-setup/debian/clock-setup.postinst
index 674fdab..29fabf9 100755
--- a/packages/clock-setup/debian/clock-setup.postinst
+++ b/packages/clock-setup/debian/clock-setup.postinst
@@ -7,6 +7,13 @@ log() {
 	logger -t clock-setup "$@"
 }
 
+# On s390 we're not allowed to change the system clock; just run tzsetup
+case "$(archdetect)" in
+    s390/*)
+	tzsetup
+	exit 0 ;;
+esac
+
 db_input medium clock-setup/ntp || true
 if ! db_go; then
 	exit 10 # back to main menu
diff --git a/packages/clock-setup/debian/control b/packages/clock-setup/debian/control
index 59da5ef..6a778c2 100644
--- a/packages/clock-setup/debian/control
+++ b/packages/clock-setup/debian/control
@@ -7,8 +7,8 @@ Build-Depends: debhelper (>= 4.1.13), po-debconf
 Vcs-Svn: svn://svn.debian.org/d-i/trunk/packages/clock-setup
 
 Package: clock-setup
-Architecture: alpha amd64 arm armel armeb hppa i386 ia64 lpia m68k mips mipsel powerpc sparc kfreebsd-i386 kfreebsd-amd64
-Depends: ${misc:Depends}, localechooser, tzsetup-udeb (>= 1:0.18), configured-network, rdate-udeb (>= 1:1.1.3-2), di-utils (>= 1.66)
+Architecture: any
+Depends: ${misc:Depends}, localechooser, tzsetup-udeb (>= 1:0.18), di-utils (>= 1.66), ${rdate:Depends}
 XC-Package-Type: udeb
 XB-Installer-Menu-Item: 2600
 Description: set up clock
diff --git a/packages/clock-setup/debian/rules b/packages/clock-setup/debian/rules
index 82ce1c1..5e3bf4b 100755
--- a/packages/clock-setup/debian/rules
+++ b/packages/clock-setup/debian/rules
@@ -1,5 +1,13 @@
 #!/usr/bin/make -f
 
+ARCH=$(shell dpkg-architecture -qDEB_HOST_ARCH)
+
+ifneq ($(ARCH), s390)
+RDATE_DEPENDS = configured-network, rdate-udeb
+else
+RDATE_DEPENDS =
+endif
+
 build:
 
 clean:
@@ -11,7 +19,9 @@ install: build
 	dh_testdir
 	dh_testroot
 	dh_clean -k
+ifneq ($(ARCH), s390)
 	dh_install finish-install.d usr/lib
+endif
 
 binary-arch: build install
 	dh_testdir
@@ -21,7 +31,7 @@ binary-arch: build install
 	dh_compress
 	dh_fixperms
 	dh_installdeb
-	dh_gencontrol
+	dh_gencontrol -- -Vrdate:Depends="$(RDATE_DEPENDS)"
 	dh_md5sums
 	dh_builddeb
 
